engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -205,27 +226,33 @@ func (pool *EnclavePool) destroyPreviousIdleEnclaves() error {
 // run method is in charge of filling the enclave pool channel with new idle enclaves
 func (pool *EnclavePool) run() {
 
-	defer close(pool.idleEnclavesChan)
-
 	fails := 0
 
 	for {
-		newEnclaveInfo, err := pool.createNewEnclave()
-		if err != nil {
-			// Retry strategy if something fails
-			logrus.Errorf("An error occurred creating a new idle enclave. Error\n%v", err)
-			fails++
-			if fails >= maxRetryAfterFails {
-				logrus.Debugf("The enclave pool sub-routine has been canceled after reaching the max retries value '%v'", maxRetryAfterFails)
-				break
+		select {
+		case <-pool.closeChan:
```

Comment:
Does this really work??? O.o 
Below, the call `pool.idleEnclavesChan <- enclaveUUID` should be blocking right? So, if the code is blocked at this line when a value is pushed to `closeChan`, there is no reason why we would end up in this `case`. Am I misunderstanding something here?

Reply:
I've changed the logic